### DISCUSSION TITLE
A mod that randomizes approach rate for each individual hitobject

### DISCUSSION THREAD

**Discussion by CenTdemeern1** [ppy/osu#17935] [Ideas]
Idea:
A mod that randomizes the approach rate for each individual hitobject, in some configurable relative range from the original AR

**Comment by peppy** [comment]
Per object random sounds potentially a bit too chaotic.. would need to see and try a proof-of-concept to make a call on whether this is something we want or not.

**Reply by PercyDan54** [reply]
I think random every combo / break section is more playable

**Reply by CenTdemeern1** [reply]
I tried it, it's quite fun and not *too* chaotic in my opinion (I have the randomness turned up to eleven in this example though)
It's definitely somewhat chaotic, but it's really not that bad during actual gameplay.
[Video footage of the proof of concept](https://youtu.be/v3JFi7VLbjk)
The pop-in should be mitigated though.
The randomness is enough to throw you off but not enough to completely screw you over.

**Reply by PercyDan54** [reply]
> I tried it, it's quite fun and not _too_ chaotic in my opinion (I have the randomness turned up to eleven in this example though) It's definitely somewhat chaotic, but it's really not that bad during actual gameplay. [Video footage of the proof of concept](https://youtu.be/v3JFi7VLbjk) The pop-in should be mitigated though. The randomness is enough to throw you off but not enough to completely screw you over.

Low star maps have low density that's why it looks not so chaotic.
Try some harder maps and you'll see it is a bit harder to time/read streams
As in this video: https://youtu.be/lxqYi7QfBrM
But overall it's not that chaotic, I still prefer random per combo, I could try to make a video if needed

**Reply by peppy** [reply]
I'd propose not 100% random but adjusting gradually over time.

**Reply by CenTdemeern1** [reply]
Like I said it was never meant to actually be 100% random anyway, the range is meant to be configurable.
But adjusting over time sounds good!
Maybe have it adjust depending on the combo?
A bit like how the muted mod does it

Also, maybe "Random per combo" could be an option in the mod configuration
